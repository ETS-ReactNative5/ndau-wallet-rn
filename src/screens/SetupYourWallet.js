import React, { Component } from 'react';

import { StyleSheet, View, ScrollView, Text, SafeAreaView } from 'react-native';
import CommonButton from '../components/CommonButton';
import Stepper from '../components/Stepper';
import cssStyles from '../css/styles';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { pushSetup } from '../actions/NavigationActions';

class SetupYourWallet extends Component {
  constructor(props) {
    super(props);
    this.state = {};

    this.props.navigator.toggleNavBar({
      to: 'hidden',
      animated: false
    });
  }

  showNextSetup = () => {
    this.props.pushSetup('ndau.SetupSeedPhrase', this.props.navigator);
  };

  render() {
    return (
      <SafeAreaView style={styles.safeContainer}>
        <View style={cssStyles.container}>
          <ScrollView style={styles.contentContainer}>
            <Stepper screenNumber={4} />

            <View>
              <Text style={cssStyles.wizardText}>
                Next we will give you a seed phrase which is the key to restoring your wallet. You
                must WRITE IT DOWN and store it in a secure location or risk losing access to your
                funds. Do not save this phrase on your device or in the cloud. Do not do this step
                in a public place.
              </Text>
            </View>
          </ScrollView>
          <View style={styles.footer}>
            <CommonButton onPress={this.showNextSetup} title="Get my seed phrase" />
          </View>
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  safeContainer: {
    flex: 1,
    backgroundColor: '#1c2227'
  },
  button: {
    marginTop: 0
  },
  text: {
    color: '#ffffff',
    fontSize: 22,
    fontFamily: 'TitilliumWeb-Regular'
  },
  contentContainer: {
    flex: 1 // pushes the footer to the end of the screen
  },
  footer: {
    justifyContent: 'flex-end'
  },
  progress: {
    paddingTop: 15,
    paddingBottom: 15
  }
});

const mapStateToProps = (state) => {
  return {};
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({ pushSetup }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(SetupYourWallet);
