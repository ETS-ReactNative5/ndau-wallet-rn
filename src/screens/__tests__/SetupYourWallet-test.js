import React from 'react';
import { StyleSheet } from 'react-native';
import SetupYourWallet from '../SetupYourWallet';

import renderer from 'react-test-renderer';

describe('testing SetupYourWallet...', () => {
  beforeEach(() => {});
  it('renders correctly', () => {
    var styles = StyleSheet.create({
      wizardText: {
        color: '#ffffff',
        fontSize: 20
      }
    });

    const tree = renderer.create(<SetupYourWallet parentStyles={styles} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
