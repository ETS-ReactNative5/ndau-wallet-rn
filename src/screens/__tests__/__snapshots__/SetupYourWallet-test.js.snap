// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing SetupYourWallet... renders correctly 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#1c2227",
      "flex": 1,
    }
  }
>
  <View>
    <RCTScrollView
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <Image
          resizeMode="contain"
          screenNumber={4}
          source={
            Object {
              "testUri": "../../../img/step-4.png",
            }
          }
          style={
            Object {
              "backgroundColor": "transparent",
              "height": 60,
              "marginBottom": 5,
              "width": "100%",
            }
          }
        />
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#ffffff",
                "fontSize": 20,
              }
            }
          >
            Next we will give you a seed phrase which is the key to restoring your wallet. You must WRITE IT DOWN and store it in a secure location or risk losing access to your funds. Do not save this phrase on your device or in the cloud. Do not do this step in a public place.
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        accessible={true}
        focusAnimation={null}
        focusAnimationDuration={225}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 4,
            "right": 4,
            "top": 6,
          }
        }
        onLayout={[Function]}
        onPress={[Function]}
        onPressIn={[Function]}
        onPressOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
        rippleCentered={false}
        rippleColor="#ffffff"
        rippleDuration={400}
        rippleFades={true}
        rippleOpacity={0.3}
        rippleSequential={true}
        rippleSize={0}
        style={
          Object {
            "backgroundColor": "#4e957a",
            "borderRadius": 2,
            "height": 36,
            "justifyContent": "space-around",
            "minWidth": 64,
            "paddingHorizontal": 8,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "fontSize": 14,
                "fontWeight": "500",
                "textAlign": "center",
              },
              Object {
                "color": "#ffffff",
                "fontFamily": "TitilliumWeb-Regular",
                "fontSize": 14,
              },
              Object {
                "color": "#ffffff",
              },
            ]
          }
        >
          GET MY SEED PHRASE
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 2,
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 2,
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
